public/src/client/topic.js

```diff
@@ -197,10 +197,21 @@ define('forum/topic', [
 			}
 		}
 
-		var currentBookmark = localStorage.getItem('topic:' + ajaxify.variables.get('topic_id') + ':bookmark');
-
-		if (!currentBookmark || parseInt(postIndex, 10) >= parseInt(currentBookmark, 10)) {
-			localStorage.setItem('topic:' + ajaxify.variables.get('topic_id') + ':bookmark', postIndex);
+		var bookmarkKey = 'topic:' + ajaxify.variables.get('topic_id') + ':bookmark';
+		var currentBookmark = localStorage.getItem(bookmarkKey);
+//		console.log('localStorage.getItem: current bookmark is ' + currentBookmark);
+
+		if (!currentBookmark || parseInt(postIndex, 10) > parseInt(currentBookmark, 10)) {
```

Comment:
I'm not sure I follow, there's an x button to close the alert, and as I understand this, it's only removing the alert if you scroll to (well, past, with this change) the last post you read. 

If you mean it won't close when you click on it, then there should be logic to close it on clock, not just when you scroll to it.

At any rate I'm hoping this alert won't be needed (I honesty don't understand the use case), but there are other ways to figure out how to close that alert box aside from constantly removing it on every scroll